pkg/tfbridge/diff.go

```diff
@@ -127,21 +126,33 @@ func visitPropertyValue(
 			}
 
 			en := name + "." + ti
-			visitPropertyValue(ctx, en, ep, e, etfs, eps, rawNames, visitor)
+			visitPropertyValue(ctx, en, ep, e, etfs, eps, visitor)
 		}
 	case v.IsObject():
-		var tfflds shim.SchemaMap
 		if tfs != nil {
-			if res, isres := tfs.Elem().(shim.Resource); isres {
-				tfflds = res.Schema()
+			if res, ok := tfs.Elem().(shim.Resource); ok {
+				tfflds := res.Schema()
+				var psflds map[string]*SchemaInfo
+				if ps != nil {
+					psflds = ps.Fields
+				}
+
+				for k, e := range v.ObjectValue() {
+					var elementPath string
+					if strings.ContainsAny(string(k), `."[]`) {
+						elementPath = fmt.Sprintf(`%s.["%s"]`, path, strings.ReplaceAll(string(k), `"`, `\"`))
```

Comment:
Umm.. How is elementPath used? Should this be a type distinct from string that encapsulates stepping down like this?

Reply:
I'm pretty sure it should be a `resource.PropertyPath`. I didn't change it because I didn't want to expand the scope of this PR any more. I'll stack another PR on top of this to refactor `elementPath`.

Reply:
Hm yeah.